lms/djangoapps/experiments/views.py

```diff
@@ -40,4 +51,32 @@ def create_or_update(self, request, *args, **kwargs):
             except ExperimentData.DoesNotExist:
                 pass
 
+        self.action = 'create'
         return self.create(request, *args, **kwargs)
+
+    def _cache_users(self, usernames):
+        users = User.objects.filter(username__in=usernames)
+        self._cached_users = {user.username: user for user in users}
+
+    def _get_user(self, username):
+        user = self._cached_users.get(username)
+
+        if not user:
+            user = User.objects.get(username=username)
+            self._cached_users[username] = user
+
+        return user
+
+    @list_route(methods=['put'], permission_classes=[permissions.IsAdminUser])
+    def bulk_upsert(self, request):
```

Comment:
Do we need to be explicit here about transaction handling? I would expect all of the records to be updated in a single transaction.